AlphaWallet/WalletConnect/WalletConnectServer.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  WalletConnectSession.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 02.07.2020.
+//
+
+import UIKit
+import WalletConnectSwift
+
+enum WalletConnectSessionError: Error {
+    case invalidWCURL
+}
+
+protocol BaseRequestHandlerDelegate: class {
+    func handler(_ handler: WalletConnectRequestHandler, didAskToSign request: Request, message: String, completion: @escaping (Bool) -> Void)
+}
+
+protocol WalletConnectRequestHandler: RequestHandler {
+    var wallet: Wallet { get set }
+    var delegate: BaseRequestHandlerDelegate? { get set }
+}
+
+protocol WalletConnectServerDelegate: class {
+    func server(_ server: WalletConnectServer, shouldConnectToSession session: String, completion: @escaping (Bool) -> Void)
+    func server(_ server: WalletConnectServer, didConnect session: String)
+    func server(_ server: WalletConnectServer, didDisconnect session: String)
+}
+
+class WalletConnectServer : NSObject {
+
+    private let walletMeta = Session.ClientMeta(name: "AlphaWallet", description: nil, icons: [], url: Config.gnosisURL)
+    private lazy var server: Server = Server(delegate: self)
+    private var requestHandlers: [WalletConnectRequestHandler] = []
+
+    dynamic var sessions: [Session] = []
```

Comment:
Does this need to be dynamic?

Reply:
now it doesn't, i was need to handle `sessions` updates, just missed to remove it.